src/main/java/ru/hogwarts/school/controller/FacultyController.java

```diff
@@ -0,0 +1,58 @@
+package ru.hogwarts.school.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import ru.hogwarts.school.model.Faculty;
+import ru.hogwarts.school.model.Student;
+import ru.hogwarts.school.service.FacultyService;
+
+import java.util.Collection;
+import java.util.List;
+
+@RestController
+@RequestMapping("/faculty")
+public class FacultyController {
+    @Autowired
+    private final FacultyService facultyService;
+
+    public FacultyController(FacultyService facultyService) {
+        this.facultyService = facultyService;
+    }
+
+    @GetMapping("/{id}")
+    public Faculty getFacultyInfo(@PathVariable("id") Long id) {
+        return facultyService.findFaculty(id);
+    }
+
+    @PostMapping
+    public Faculty createFaculty(@RequestBody Faculty faculty) {
+
+        return facultyService.createFaculty(faculty);
+    }
+
+    @PutMapping
+    public ResponseEntity<Faculty> editFaculty(@RequestBody Faculty faculty) {
+        Faculty foundFaculty = facultyService.editFaculty(faculty);
+        if (foundFaculty == null) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).build();
+        }
+        return ResponseEntity.ok(foundFaculty);
+    }
+
+    @DeleteMapping("{id}")
+    public ResponseEntity<Faculty> deleteFaculty(@PathVariable Long id) {
+        facultyService.deleteFaculty(id);
+        return ResponseEntity.ok().build();
+
+    }
+    @GetMapping
+    public List<Faculty> findByColor(@RequestParam String color) {
+        return facultyService.findColor(color);
+    }
+    @GetMapping("/all")
```

Comment:
```suggestion
    @GetMapping
```
никаких слов быть не должно

Reply:
этого нет в задании, я добавил для удобства, чтобы вывести весь список

Reply:
я понял, но комментарий по-прежнему актуальный - путь должен быть просто /, без /all, т.к. это нарушает REST